### DISCUSSION TITLE
Ansys 2024R2 Geometry Service not installed.

### DISCUSSION THREAD

**Discussion by dkowollik** [ansys/pyansys-geometry#1374] [Q&A] [answered]
Hi @RobPasMue,
I tried to follow your instructions on the doc pages to install the GeometryService. PyAnsys with SpaceClaim, Discovery and PyPrimeMesh is working fine to load geometries. I want to use the GeometryService like this:
`from ansys.geometry.core import launch_modeler_with_geometry_service`
`modeler = launch_modeler_with_geometry_service()`
The Feedback is a FileNotFoundError. I debugged into the launch method. 
![grafik](https://github.com/user-attachments/assets/946c375a-bb9a-498a-b94b-4e9157039592)
The folder GeometryService does not exist in the Ansys installation folder. Therefore, I tried to install it with `setup.exe -silent -geometryservice` like it is described in your documentation. The structure bundle I have does not accept the parameter `-geometryservice`. Therefore, I reinstalled the entire ANSYS 2024R2 release with the AnsysInstaller.exe, which is supposed to install the GeometryService bei default. This did not work. the folder is still missing. What am I missing? Is there a different bundle, I have to download? Can you please help out?

**Comment by RobPasMue** [comment, marked as answer]
Hi @dkowollik! I believe you are not alone on this issue - I have also experienced it myself. Currently the flag ``-geometryservice`` is not working properly. We have reported this problem to the Unified Installer team and they should try and fix it soon. However, to prevent it from blocking your installation, let me give you some steps to follow:

1. Access the Ansys Customer Portal at https://download.ansys.com/Current%20Release
2. Download the Ansys 2024 R2 ISO image **1** (only the first one is required) - see screenshot
![image](https://github.com/user-attachments/assets/16eaf196-3080-4c44-a1f1-2033c6705519)
3. Open the ISO image and navigate to the current folder on the screenshot (in my case the ISO image was mounted on drive D:/ so... ``D:\geometryservice``)
![image](https://github.com/user-attachments/assets/ac417a98-44e6-4dd0-b1ed-b4f50b22bdd5)
4. Copy this WINX64.7z zip file to your current Ansys Installation folder (i.e. by default it should be ``C:\Program Files\ANSYS Inc\v242``)
![image](https://github.com/user-attachments/assets/26312ffc-d122-42bc-9ec9-bb2f780f7fbe)
5. Uncompress the zip file and extract it to ``C:\Program Files\ANSYS Inc\v242`` like in this screenshot
![image](https://github.com/user-attachments/assets/8b980968-a02d-403d-b0ba-2b18a2832a8c)
6. Now, you should have a folder called ``C:\Program Files\ANSYS Inc\v242\GeometryService`` 🥳 

Let me know if you encounter any issues in the process! I'm sorry for the workaround but at least it will unblock your situation.

**Reply by dkowollik** [reply]
Hi @RobPasMue,
thank you for the quick response. I will download the image at lunch time, give it a try and give you a feedback.

**Reply by RobPasMue** [reply]
Awesome! Let me know how it goes. Happy to help if you need any support.

**Reply by dkowollik** [reply]
Hi @RobPasMue,
now I have the folder and it seemed to start once. I restarted the terminal and now I get the error: `ConnectionError: Timeout while waiting for backend to be ready.`
I might not have executed `modeler.exit()`. Do I have to kill a process in the task manager to fix this?

**Reply by RobPasMue** [reply]
Ah probably... modeler.exit() is important I believe because otherwise the service will continue to run... It might be showing up in the task manager. Another option is that you restart your machine.

Nonetheless, the command would be ``modeler.close()`` to shut down the service. I think that, in order to be aligned with other packages, I am going to implement ``modeler.exit()`` as well. They will basically do the same thing though.

**Reply by RobPasMue** [reply]
See https://github.com/ansys/pyansys-geometry/pull/1375 - working on including ``modeler.exit()``

**Reply by dkowollik** [reply]
How do I find the process that I need to kill? Do you have a name or handle that I can search for?

**Reply by dkowollik** [reply]
`modeler.close()` is fine for me. I mixed it up with the traditional python exit() function. Other pyansys servers such as PyPrimeMesh seem to also shut down once the python script is finished or if one uses exit() in an ipython terminal.

**Reply by RobPasMue** [reply]
Yes - I just debugged it and it should show up something like this

![image](https://github.com/user-attachments/assets/754dd723-a52e-4ef8-9b0a-f9e12f3341b0)

I launched it the same way you did and then looked into the PID associated to the process (see the debugger screenshot). Then I searched for it in the Task Manager. In your case since your script finished you no longer have access to the PID but you should be able to search for the background process.

![image](https://github.com/user-attachments/assets/88f3e632-fd93-44de-8541-163840c45267)

**Reply by RobPasMue** [reply]
FYI - it might also show up like this in the Task Manager

![image](https://github.com/user-attachments/assets/bd50ca05-889f-4ee4-a1f7-1d287a1f3455)

**Reply by dkowollik** [reply]
I could not find the process, restarted my computer and still have the same timeout error. Do you have another idea for the response?

**Reply by RobPasMue** [reply]
I have a question first though. Are you on a VPN environment? This might be causing issues (potentially).

Nonetheless, we can have a look at the logs as well. Whenever you launch the Geometry Service, you should get some logs in the following folder: ``C:\Program Files\ANSYS Inc\v242\GeometryService\Logs``. Could you have a look there and see if you see anything interesting in the last executions?

**Reply by dkowollik** [reply]
The log file was last edited at 12:49 today. I guess this was the first and only time the server started. The last line in the log file is: 
Server started. Listening on endpoint: localhost:61780

**Reply by dkowollik** [reply]
I do not use VPN today.

**Reply by RobPasMue** [reply]
Hmmm... Can you try running your script from an "admin" terminal? Or open your IDE in admin mode and try running it too?

**Reply by dkowollik** [reply]
I did try ipython within "admin" mode and it worked. Afterwards, I changed the read/wirte permissions on `C:\Program Files\ANSYS Inc\v242\GeometryService\Logs` to all users,  tried again without "admin" mode and it also worked to start the modeler. May be it makes more sense to track the log files in the user directory like the other ansys processes do: 
`C:\Users\%username%\AppData\Roaming\Ansys\v242`   
Do I have to change access permissions to any other folder in GeometryServices as well?

**Reply by RobPasMue** [reply]
Aha so that was it! I'm glad you figured it out. No, I think that's the only folder you need to change the permissions on. Mostly because the logging takes place there with the Geometry Service. BTW, your suggestion makes total sense so I will incorporate it soon! I really like the idea of having it in ``C:\Users\%username%\AppData\Roaming\Ansys\v242``. Let me work on that!

Also, if my initial response helped you out to install the Geometry Service (which is the original request of the discussion), I would kindly ask you to click on "Mark as answer" on the first comment of this thread so that other users can benefit from it!

I appreciate all your time and effort to get it working @dkowollik! Your experience will definitely help other users onboard PyAnsys Geometry!

**Reply by susannehaase** [reply]
I also stumbled over the log folder permission thing. Solution: 
`modeler = launch_modeler_with_geometry_service(server_logs_folder = r"D:\Temp")`
Regards, Susanne

**Reply by RobPasMue** [reply]
Hi @susannehaase! Thanks for commenting on this :) If you are using version 0.7.1 or above you might want to know that the server logs for the Geometry Service are now stored by default in ``` %PUBLIC%/Documents/Ansys/GeometryService ``` (unless specified differently via argument using ``server_logs_folder``). Users shouldn't have problems to access this location from now onwards =)

https://github.com/ansys/pyansys-geometry/blob/ef3af381f2f17a0bb65c81081b03cc1afafff3c2/src/ansys/geometry/core/connection/launcher.py#L591-L596
